src/server/routes/pathways/generate-pathway-json/biopax-metadata/index.js

```diff
@@ -1,226 +1,145 @@
-
-    //Get Chemical Formula
-    let formula = biopaxElement['chemicalFormula'];
-    if (formula) { result.push(['Chemical Formula', formula]); }
-  }
-
-  //Get database id
-  let db = biopaxElement['db'];
-  let id = biopaxElement['id'];
-  if (db && id) { result.push(['Database ID', [db, id]]); }
-
-  return result;
-}
-
-/**
- *
- * @param {*} entity Metadata for an entity in the network
- * @returns Tree array containing all metadata about the given entity
- */
-function buildBiopaxTree(entity) {
-  let result = [];
-
-  //make sure the metadata exists
-  if (!(entity))
-    return;
-
-  //Get the node id, as defined in getProcessedBioPax
-  let id = entity['pathid'];
-  if (!(id)) { id = entity['about']; }
-  if (!(id)) { return; }
-
-  //get BioPAX metadata for this node
-  let collectedData = collectEntityMetadata(entity);
-
-  //Collect Data for cross-references
-  let xrefList = [];
-  let xref = entity['xref'];
-  let eref = entity['entityReference'];
-  if(xref)
-    if(typeof xref === 'string')
-      xrefList = _.concat(xrefList,[xref]);
-    else
-      xrefList = _.concat(xrefList,xref);
-  if(eref)
-    xrefList = _.concat(xrefList,[eref]);
-
-  processXrefs(xrefList,eref,collectedData);
-
-  result.push([id,collectedData]);
-  return result;
-}
-
-function processXrefs(xrefList,eref,collectedData){
-  let erefXrefList = [];
-
-  for (let i = 0; i < xrefList.length; i++) {
-    //Check if this is a cross-reference or entity reference
-    //(- the last item (if eref is defined) is treated in a special way that
-    //TODO: saves a couple of lines but sacrifices clarity...)
-    let keyName = 'Reference';
-    if (i == (xrefList.length - 1) && eref)
-      keyName = 'EntityReference';
-
-    //Get Referenced element and make sure it's valid
-    let refElement = getByNodeId(xrefList[i], biopaxMap);
-    if (!(refElement)) {
-      continue;
+let extractBiopaxMetadata = biopaxJsonEntry => {
+  let databaseIds = _.get(biopaxJsonEntry, 'dbIds', []);
+  let entRefEl = _.get(biopaxJsonEntry, 'entRefEl', []);
+
+  let type = _.get(biopaxJsonEntry, '@type', '');
+  let datasource = _.get(biopaxJsonEntry, 'dataSource', '');
+  let comments = _.get(biopaxJsonEntry, 'comment', []);
+
+
+  let entryDisplayName = _.get(biopaxJsonEntry, 'displayName', '');
+  let entryNames = [].concat(_.get(biopaxJsonEntry, 'name', []));
+  let entryStdName = _.get(biopaxJsonEntry, 'standardName', '');
+
+  let entRefNames = _.get(entRefEl, 'name', []);
+  let entRefStdName = _.get(entRefEl, 'standardName', '');
+  let entRefDisplayName = _.get(entRefEl, 'displayName', '');
+
+
+  let synonyms = ( 
+    _.uniq( 
+      _.flatten( [entryNames, entRefNames] ) 
+    ).filter( name => !_.isEmpty( name ) )
+  );
+
+  let standardName = entryStdName || entRefStdName || '';
+  let displayName = entryDisplayName || entRefDisplayName || '';
+
+  return {
+    comments,
+    synonyms,
+    datasource,
+    type: type.includes(':') ? type.split(':')[1] : type,
```

Comment:
- maybe add a comment here explaining the type processing e.g. 'bp:Protein' -> 'Protein'
- also TODO need to split camel case type e.g. 'BiochemicalReaction' -> 'Biochemical Reaction'